core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthChecker.java

```diff
@@ -89,10 +132,36 @@ private synchronized void check() {
 
     @Override
     public synchronized CompletableFuture<?> closeAsync() {
-        if (lastCheckFuture != null) {
-            return lastCheckFuture.handle((unused1, unused2) -> null);
+        if (lastResponse != null) {
+            lastResponse.abort();
+            return lastResponse.completionFuture().handle((unused1, unused2) -> null);
```

Comment:
I think we still have a chance that the `check` method is invoked.
If the task is scheduled by `executor.schedule(this::check, nextDelayMillis, TimeUnit.MILLISECONDS);`, and this method is called, the scheduled task will be executed.
How about introducing a flag for that?

Reply:
Added the `closed` flag.